list-collections/arraylist/src/main/java/ua/hillel/ArrayList.java

```diff
@@ -0,0 +1,156 @@
+package arraylist.src.main.java.ua.hillel;
+
+import common.src.main.java.ua.hillel.List;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Implement a list that is based on array. This resizable data structure
+ * is based on an array and is a simplified version of {@link java.util.ArrayList}.
+ */
+public class ArrayList<T> implements List<T> {
+
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    Object[] elementData;
+    int size;
+
+    public ArrayList() {
+        this.elementData = EMPTY_ELEMENTDATA;
+    }
+
+    @Override
+    public void add(int index, T element) {
+        rangeCheckForAdd(index);
+        final int s;
+        Object[] elementData;
+        if ((s = size) == (elementData = this.elementData).length)
+            elementData = grow();
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                s - index);
+        elementData[index] = element;
+        size = s + 1;
+    }
+
+    @Override
+    public void add(int index, T element, int size) {
+
+    }
+
+    @Override
+    public T get(int index) {
+        Objects.checkIndex(index, size);
+        return elementData(index);
```

Comment:
method `elementData` is redundant. Remove it and replace it with a call to the internal array.